src/Microsoft.Identity.Web.TokenAcquisition/MicrosoftIdentityMessageHandler.cs

```diff
@@ -0,0 +1,481 @@
+                throw new MicrosoftIdentityAuthenticationException(
+                    "Authentication options must be configured either in default options or per-request using WithAuthenticationOptions().");
+            }
+
+            // Get scopes from options
+            var scopes = options.Scopes;
+            
+            if (scopes == null || !scopes.Any())
+            {
+                throw new MicrosoftIdentityAuthenticationException(
+                    "Authentication scopes must be configured in the options.Scopes property.");
+            }
+
+            // Send the request with authentication
+            var response = await SendWithAuthenticationAsync(request, options, scopes, cancellationToken).ConfigureAwait(false);
+
+            // Handle WWW-Authenticate challenge if present
+            if (response.StatusCode == HttpStatusCode.Unauthorized)
+            {
+                string? challengeClaims = null;
+                
+                // Try to extract claims from WWW-Authenticate header if available
+                if (response.Headers.WwwAuthenticate != null)
+                {
+                    foreach (var authenticateHeader in response.Headers.WwwAuthenticate)
+                    {
+                        var headerValue = authenticateHeader.ToString();
+                        if (headerValue.StartsWith("Bearer", StringComparison.OrdinalIgnoreCase))
+                        {
+                            // Look for claims parameter in the Bearer challenge
+                            // Format: Bearer realm="...", claims="base64encodedclaims"
+                            var claimsMatch = ClaimsExtractionRegex.Match(headerValue);
+                            
+                            if (claimsMatch.Success)
+                            {
+                                challengeClaims = claimsMatch.Groups[1].Value;
+                                break;
+                            }
+                        }
+                    }
+                }
+                
+                if (!string.IsNullOrEmpty(challengeClaims))
+                {
+                    _logger?.LogInformation(
+                        "Received WWW-Authenticate challenge with claims. Attempting token refresh.");
+
+                    // Create a new options instance with the challenge claims
+                    var challengeOptions = CreateOptionsWithChallengeClaims(options, challengeClaims);
+                    
+                    // Clone the original request for retry
+                    using var retryRequest = await CloneHttpRequestMessageAsync(request).ConfigureAwait(false);
+                    
+                    try
+                    {
+                        // Attempt to get a new token with the challenge claims
+                        var retryResponse = await SendWithAuthenticationAsync(retryRequest, challengeOptions, scopes, cancellationToken).ConfigureAwait(false);
```

Comment:
Add some logging on the retry response. particularly if it is still 401 and if it has WWW-Authenticate.